### DISCUSSION TITLE
/cmi setblock

### DISCUSSION THREAD

**Discussion by ItLucifer-MRPG** [Zrips/CMI#5681] [Feature Requests]
A cmi version of /setblock where you can add more details, like specific worlds for a specific amount of time, etc 

Ive been using alias to set a specific block to be a redstone block to allow alias work to enable command block but ive noticed that it only works if you're in the specific world with the command block

 I feel like a command like this will make things like that alot easier

**Comment by CloudeLecaw** [comment]
nice idea.. but pls not only worrld args

` /setblock [material:meta] (from x,y,z) (to x,y,z) (world) (-s)`

this might be usefull..
if no cords are given the actual cord is taken to place the material there

Problem might be - this would need a loader too because u cant garantee the blocks in the worlds or area where u wanted them are loaded.
so this also would need a chunkloader/unloader part to place the blocks in the area if chunk is not active at this moment)
welll...may thats the reason why worldedit and also vanillas fill command dont do this..

Well until at setblock might be added one day you can use vanillas fill command.
fill fromPos toPos material. Note - also here you have to be in that world... may the loaded chunks thing is the reason?.....

_example:   fills area from -138 63 248 to -149 65 229 with stone_
`minecraft:fill -138 63 248 -149 65 229 minecraft:stone`

we are using this fill command in some areas where we allow players to "reset testareas" (they need to be in the world where the command (attached to an IC in our case) is activeated.


Note: Not even the famous WorldEdit could solve this worldproblem...what you think why...?
